spring-cloud-gcp-storage/src/main/java/org/springframework/cloud/gcp/storage/GoogleStorageResourceBucket.java

```diff
@@ -110,11 +125,16 @@ public long lastModified() {
 	}
 
 	@Override
-	public Resource createRelative(String relativePath) throws IOException {
-		return new GoogleStorageResourceObject(this.storage,
+	public GoogleStorageResourceObject createRelative(String relativePath) throws IOException {
+		GoogleStorageResourceObject resource = new GoogleStorageResourceObject(this.storage,
 				getURI().toString() + (relativePath.startsWith("/") ? "" : "/")
 						+ relativePath,
-				this.createBucketIfRequired);
+				this.createBucketIfNotExists);
+		if (!resource.exists()) {
+			resource.create();
```

Comment:
Basically the same comment I had on the other class: perhaps just pass `true` in the constructor for the relative blob? It looks like `resource.create()` will either result in a blob that is created and exists, or will throw an exception, so passing `this.createBucketIfNotExists` in the constructor doesn't have any meaning anymore.